packages/yourloops/pages/hcp/team-edit-dialog.tsx

```diff
@@ -136,42 +145,53 @@ function TeamEditModal(props: TeamEditModalProps): JSX.Element {
   });
 
   const isFormIsIncomplete = (): boolean => {
-    let valid = teamName.length > 1;
-    valid = valid && teamPhone.length > 1;
-    valid = valid && addrLine1.length > 1;
-    valid = valid && addrZipCode.length > 1;
-    valid = valid && addrCity.length > 1;
-    valid = valid && addrCountry.length > 1;
-    valid = valid && (teamEmail.length < 1 || REGEX_EMAIL.test(teamEmail));
+    const inLimit = (value: string, limits: { min: number, max: number }): boolean => {
+      const len = value.length;
+      return len > limits.min && len < limits.max;
+    };
+    let valid = inLimit(teamName.trim(), teamFieldsLimits.name);
+    valid = valid && inLimit(teamPhone.trim(), teamFieldsLimits.phone);
+    valid = valid && inLimit(addrLine1.trim(), teamFieldsLimits.addLine1);
+    valid = valid && inLimit(addrLine2.trim(), teamFieldsLimits.addLine2);
+    valid = valid && inLimit(addrZipCode.trim(), teamFieldsLimits.zipCode);
+    valid = valid && inLimit(addrCity.trim(), teamFieldsLimits.city);
+    valid = valid && inLimit(addrCountry.trim(), teamFieldsLimits.country);
+
+    const email = teamEmail.trim();
+    if (valid && inLimit(email, teamFieldsLimits.email)) {
+      valid = REGEX_EMAIL.test(email);
+    }
+
     return !valid;
   };
 
   const handleCloseModal = (): void => {
-    setModalOpen(false);
+    onSaveTeam(null);
   };
   const handleValidateModal = (): void => {
-    setModalOpen(false);
+    const updatedTeam = _.cloneDeep(team);
+    updatedTeam.name = teamName.trim();
+    updatedTeam.phone = teamPhone.trim();
 
-    team.name = teamName;
-    team.phone = teamPhone;
-
-    if (teamEmail.length > 0) {
-      team.email = teamEmail;
+    const email = teamEmail.trim();
+    if (email.length > 0) {
+      updatedTeam.email = email;
     } else {
-      delete team.email;
+      delete updatedTeam.email;
     }
 
-    team.address = {
-      line1: addrLine1,
-      line2: addrLine2,
-      zip: addrZipCode,
-      city: addrCity,
-      country: addrCountry,
+    const line2 = addrLine2.trim();
+    updatedTeam.address = {
+      line1: addrLine1.trim(),
+      line2,
+      zip: addrZipCode.trim(),
+      city: addrCity.trim(),
+      country: addrCountry.trim(),
     };
-    if (addrLine2.length < 1) {
-      delete team.address?.line2;
+    if (line2.length < 1) {
+      delete updatedTeam.address.line2;
     }
-    onSaveTeam(team);
+    onSaveTeam(updatedTeam);
   };
   const handleChangeTeamName = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
```

Comment:
```suggestion
  
  const handleChange = (
    setState: React.Dispatch<React.SetStateAction<string>>
  ): ((event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void) => (event) => setState(event.target.value);
```

Reply:
I wanted to select lines 196 to 216

Reply:
Sorry I don't understand this complex line.
Having a function, which return a function (x2) make things hard to understand.